omaha/core/core.cc

```diff
@@ -277,9 +277,34 @@ HRESULT Core::DoMain(bool is_system, bool is_crash_handler_enabled) {
     return hr;
   }
 
-  std::unique_ptr<Scheduler> scheduler(new Scheduler(*this));
-  hr = scheduler->Initialize();
+  auto scheduler = std::make_unique<Scheduler>();
+
+  // Start update worker
+  hr = scheduler->Start(cm.GetUpdateWorkerStartUpDelayMs(),
+                        cm.GetAutoUpdateTimerIntervalMs(),
+                        [this](auto*) {
+    StartUpdateWorker();
+  });
+
+  if (FAILED(hr)) {
+    OPT_LOG(LW, (L"[Failed to start update worker scheduler][0x%08x]", hr));
+    return hr;
+  }
+
+  // Start Code Red worker
+  const int cr_timer_interval = cm.GetCodeRedTimerIntervalMs();
+  hr = scheduler->Start(cr_timer_interval,
+                        [this, cr_timer_interval](HighresTimer* debug_timer) {
+    StartCodeRed();
+    if (debug_timer) {
+      int actual_time_ms = static_cast<int>(debug_timer->GetElapsedMs());
```

Comment:
We'd need to include the definition of HighresTimer.

Reply:
I moved the include to scheduler.h, so that brings the HighresTimer definition.